#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("inscribed_radius", double_t, 0, "inscribed radius", 0.2, 0.05, 1.0)
gen.add("max_inflation_radius", double_t, 0, "max inflation radius", 0.4, 0.05, 1.0)
gen.add("roughness_threshold", double_t, 0, "roughness threshold", 0.5, 0.01, 3.1415)
gen.add("height_diff_threshold", double_t, 0, "height differences threshold", 0.4, 0.05, 1.0)
gen.add("local_radius", double_t, 0, "local radius", 0.2, 0.02, 1.0)
gen.add("riskiness_factor", double_t, 0, "riskiness factor", 1.0, 0, 1.0)
gen.add("height_diff_factor", double_t, 0, "height differences factor", 1.0, 0, 1.0)
gen.add("roughness_factor", double_t, 0, "roughness factor", 1.0, 0, 1.0)

exit(gen.generate("mesh_navigation", "nav_core", "MeshGlobalPlanner"))
